### DISCUSSION TITLE
Idiomatic way to modify config settings depending on operating system (e.g., macOS vs Linux) ?

### DISCUSSION THREAD

**Discussion by miguno** [wezterm/wezterm#4728] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20230712-072601-f4abf8fd

### Ask your question!

My goal is to change, for example, the keyboard modifier in my wezterm configuration  (Alt vs. Opt vs. Command and so on) for my keyboard depending on whether wezterm is running on macOS vs. Linux, so that I can hit the same physical modifier key regardless of which OS I am using at that moment.

Is there an idiomatic way to do so?  I am looking at settings such as:

```lua
send_composed_key_when_left_alt_is_pressed = false, -- LOption as Alt
send_composed_key_when_right_alt_is_pressed = true, -- ROption as composed key
```
which I'd then conditionally set depending on the OS.

I hunted through the existing documentation and also searched through the Q&A here, but didn't spot something useful. Perhaps I have overlooked something obvious.

Any pointers?

**Comment by nekowinston** [comment, marked as answer]
I like to keep these functions in a separate utils file, but here's the rough idea:
```lua
local wezterm = require("wezterm")
local c = wezterm.config_builder()

local is_linux = function()
	return wezterm.target_triple:find("linux") ~= nil
end

local is_darwin = function()
	return wezterm.target_triple:find("darwin") ~= nil
end

-- simple check, true on darwin
c.send_composed_key_when_right_alt_is_pressed = is_darwin()
-- ternary, 0.8 on darwin, 1.0 elsewhere
c.window_background_opacity = is_darwin() and 0.8 or 1.0
```
If you need to, you can combine them into a `is_unix` or `is_windows` etc. etc.

**Reply by miguno** [reply]
Thanks a lot, @nekowinston !

In addition to the helpful code snippets, the pointer to `wezterm.target_triple` also helped me to discover https://wezfurlong.org/wezterm/config/lua/wezterm/target_triple.html in the documentation.

**Reply by bbkane** [reply]
Is there an advantage to making this a function instead of a const variable? I've put the following and it seems to work fine:

```lua
-- https://github.com/wez/wezterm/discussions/4728
local is_darwin <const> = wezterm.target_triple:find("darwin") ~= nil
local is_linux <const> = wezterm.target_triple:find("linux") ~= nil
```

And I'm using it like this:

```lua
if is_darwin then
        config.font_size = 18.0
end
```

**Reply by nekowinston** [reply]
There shouldn't be any difference AFAIK.

**Reply by wez** [reply]
no practical difference. If you are doing a lot of conditional checks, assigning to a local and reading that value back is less work for the underlying lua implementation than calling a function. But you would have to be making a LOT of conditional checks for that to be noticeable to a human and it seems unlikely for that to be true in a wezterm configuration.
